src/didcommv2/PackUnpack.ts

```diff
@@ -1,54 +1,119 @@
+                signFrom, 
+                protectSender,
+                attachments
+            )
+            return packed
+        } else {
+            const didDoc = await resolve(from)
+            const kid = didDoc.keyAgreement![0]
+            const _key = getItem(kid! as string) !== undefined ?  getItem(kid! as string) : await getAsyncItem(kid! as string)
+            const key = JSON.parse(_key!)
+            var privateKey = key.privateJwk
+            privateKey.kid = kid
+
+            var packed = await DIDCommModule.pack(
+                JSON.stringify(msg),
+                uuid.v4(),
+                typeof thid !== 'undefined' ? thid:null,
+                to,
+                from,
+                messageType,
+                {},
+                JSON.stringify(privateKey),
+                false,
+                false,
+                attachments
+            )
+            return packed[0]
+
+        }
+        
     } catch (error: any) {
         logger("didcomm - Pack error", error)
     }
 }
 
 export async function unpack(packMsg: any) {
     try {
-        // TODO case with several key agreement in did doc
-        console.log("RM HERE")
-        console.log(packMsg)
-        const recipient = packMsg.recipients[0].header.kid.split("#")[0]
-        const didDoc = await resolveDIDPeer(recipient)
-        const kid = didDoc.keyAgreement[0].id
-        const key = JSON.parse(await getItem(kid)!)
-        var privateKey = key.privateJwk
-        privateKey.kid = kid
-        var unpacked = await DIDCommV2Module.unpack(
-            JSON.stringify(packMsg), 
-            privateKey, 
-          )
-        return unpacked
+        if (Platform.OS === 'android') {
+            // TODO case with several key agreement in did doc
+            const recipient = packMsg.recipients[0].header.kid.split("#")[0]
+
+            const didDoc = await resolve(recipient)
+            const kid = didDoc.keyAgreement![0]
+            const _key = getItem(kid! as string) !== undefined ?  getItem(kid! as string) : await getAsyncItem(kid! as string)
+            const key = JSON.parse(_key!)
+            var privateKey = key.privateJwk
+            privateKey.kid = kid
+
+
+            var unpacked = await DIDCommV2Module.unpack(
+                JSON.stringify(packMsg), 
+                privateKey, 
+            )
+            return unpacked
+        } else {
+            const recipient = packMsg.recipients[0].header.kid.split("#")[0]
+            const didDoc = await resolve(recipient)
+            const kid = didDoc.keyAgreement![0]
+            const _key = getItem(kid! as string) !== undefined ?  getItem(kid! as string) : await getAsyncItem(kid! as string)
+            const key = JSON.parse(_key!)
+            var privateKey = key.privateJwk
+            privateKey.kid = kid
+            
+            const unpacked = await DIDCommModule.unpack(JSON.stringify(packMsg),JSON.stringify(privateKey)) 
```

Comment:
is everything in this 'else' the same as what is in the 'if' except the return statement line? if so, can the duplicate code get pulled above?

Reply:
yes!. The code needs to be improved and refactored. This PR was just the beginning :)